src/PKSim.Core/Services/HIDiseaseStateImplementation.cs

```diff
@@ -0,0 +1,186 @@
+﻿using System;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Exceptions;
+using OSPSuite.Utility.Extensions;
+using PKSim.Assets;
+using PKSim.Core.Model;
+using PKSim.Core.Repositories;
+using static PKSim.Core.CoreConstants.Units;
+using static PKSim.Core.CoreConstants.Organ;
+using static PKSim.Core.CoreConstants.Parameters;
+using IFormulaFactory = OSPSuite.Core.Domain.Formulas.IFormulaFactory;
+
+namespace PKSim.Core.Services;
+
+public class HIDiseaseStateImplementation : IDiseaseStateImplementation
+{
+   public static class ChildPughScore
+   {
+      public static double A = 0;
+      public static double B = 1;
+      public static double C = 2;
+   }
+
+   private static readonly Cache<double, double> _portalFlowScalingFactor = new()
+   {
+      {ChildPughScore.A, 0.4},
+      {ChildPughScore.B, 0.36},
+      {ChildPughScore.C, 0.04},
+   };
+
+   private static readonly Cache<double, double> _hepaticFlowScalingFactor = new()
+   {
+      {ChildPughScore.A, 1.3},
+      {ChildPughScore.B, 2.3},
+      {ChildPughScore.C, 3.4},
+   };
+
+   private static readonly Cache<double, double> _hepaticVolumeScalingFactor = new()
+   {
+      {ChildPughScore.A, 0.69},
+      {ChildPughScore.B, 0.55},
+      {ChildPughScore.C, 0.28},
+   };
+   private static readonly Cache<double, double> _renalFlowScalingFactor = new()
+   {
+      {ChildPughScore.A, 0.88},
+      {ChildPughScore.B, 0.65},
+      {ChildPughScore.C, 0.48},
+   };
+
+   public const string CHILD_PUGH_SCORE = "Child-Pugh score";
+   //TODO
+   private const int HI_VALUE_ORIGIN_ID = 92;
+
+
+   private readonly IDimension _ageDimension;
+   private readonly IValueOriginRepository _valueOriginRepository;
+   private readonly IFormulaFactory _formulaFactory;
+   public HIDiseaseStateImplementation(IDimensionRepository dimensionRepository, IValueOriginRepository valueOriginRepository, IFormulaFactory formulaFactory)
```

Comment:
@Yuri05 Just the beginning. I will create a base class for all the methods that are common. For now, I am just trying to make the test pass